etl/src/main/java/org/pih/hivmigration/etl/sql/Setup.groovy

```diff
@@ -62,6 +62,22 @@ class Setup extends SqlMigrator {
             DELIMITER ;
         ''')
 
+        executeMysql("Create function try_to_fix_date", '''
```

Comment:
Was the bug I was seeing caused by invalid date values?

Reply:
I doubt it. I expect it was caused by a faulty join. But using this function allowed simplifying the date logic substantially, eliminating those joins, and the problem.